src/SAML2/XML/IdentifiersTrait.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML;
+
+use SAML2\XML\saml\AbstractBaseIDType;
+use SAML2\XML\saml\BaseID;
+use SAML2\XML\saml\NameID;
+use SAML2\XML\saml\EncryptedID;
+
+/**
+ * Trait grouping common functionality for elements that can hold identifiers.
+ *
+ * @author Tim van Dijen, <tvdijen@gmail.com>
+ * @package simplesamlphp/saml2
+ */
+trait IdentifiersTrait
```

Comment:
The thing is we just tried to split the identifiers, because BaseIDType and NameIDType are unrelated (i.e. have no common ancestor) by spec.

Reply:
I know, I know, but that's unrelated. I mean, that common functionality was related to name-qualifiers, while this is putting together different types of IDs. The former is ok to join in a trait, but this one isn't, I think. A common interface for all ID types would be better.

Reply:
Ok, I can relate to the common interface.. I think it will probably be empty, since EncryptedID has nothing in common with the others, but that's okay..
I can still see a use for the IdentifiersTrait thought.. It could implement a setter/getter and be reused on every element that has an identifier.. Agree?

Reply:
> Ok, I can relate to the common interface.. I think it will probably be empty, since EncryptedID has nothing in common with the others, but that's okay..

Actually, I think such an interface should mandate the implementation of a `getValue()` method (or whatever we decide to call it). Of course, the value is not immediately available on an encrypted ID, **but** you could get it by decrypting first. If the encrypted ID has been provided with a decryption key, then decryption is possible and you could call `getValue()` on it.

> I can still see a use for the IdentifiersTrait thought.. It could implement a setter/getter and be reused on every element that has an identifier.. Agree?

You mean something like `setID()`/`getID()`, plus the corresponding property? Then yeah, I agree 😄

Reply:
Yes, that's what I meant..  Apart from the getValue this has all been addressed in the last commit